### PR TITLE
Issue #7: Don't exit if qemu-agent is not available

### PR DESCRIPTION
**Pull request by jlausuch**
proof run with image with qemu-agent not running on purpose: https://pastebin.com/raw/c2zLxjGy

### CODE REVIEW

**Comment by asmorodskyi** [comment]
you introduce variable which you set to True without any prove . then you using this variable inside a `execute_cmd` function to check that you can execute it. And only way to get variable to False is actually via fail to execute same function but with some "test" command. 

I would say that we should or drop this variable notion  at all and just simple throw exception when we can not execute command. Or find out some "other" way to detect availability of qemu agent

**Comment by jlausuch** [comment]
> you introduce variable which you set to True without any prove . then you using this variable inside a `execute_cmd` function to check that you can execute it. And only way to get variable to False is actually via fail to execute same function but with some "test" command.
Yes, but it is set to false if `wait_for_qemu_agent_ready()` times out the first time. The reason it's true at the beginning is cause it needs to be able to run `execute_cmd()` to verify qemu-agent is running on the VM right after deployment time.

> 
> I would say that we should or drop this variable notion at all and just simple throw exception when we can not execute command. Or find out some "other" way to detect availability of qemu agent
Would you advise to do that check every time `execute_cmd()` is called? It wouldn't be so efficient..

File: qatrfm/domain.py [hunk 1]
```diff
@@ -160,7 +165,9 @@ def wait_for_qemu_agent_ready(self, timeout=300):
             except libutils.TrfmCommandFailed:
                 i += 1
                 time.sleep(10)
-        raise libutils.TrfmDomainTimeout
+        self.logger.warning("Qemu-agent is not available on the domain {}."
+                            .format(self.name))
+        self.qemu_agent = False
```
Comment (cfconrad) [inline qatrfm/domain.py]: I would raise that exception as well. So this method acts always the same first and all others.

```
raise libutils.TrfmQemuAgentNotReady("Qemu-agent is not running on the domain")
```

I'm not sure if we should try qemu_agent always or only the first time. Is there a possibility that qemu_agent can be started afterwards?

Comment (jlausuch) [reply qatrfm/domain.py]: The purpose of the PR is not to exit if qemu-agent is not running in the VM, which might be the case for some images (e.g. non-openqa images)

Comment (jlausuch) [reply qatrfm/domain.py]: > I'm not sure if we should try qemu_agent always or only the first time. Is there a possibility that qemu_agent can be started afterwards?

It is actually possible, just checked, but I don't think it's desired and users will really do that, but it's possible

    systemctl start qemu-ga@virtio\\\\x2dports-org.qemu.guest_agent.0.service

Comment (asmorodskyi) [reply qatrfm/domain.py]: > The purpose of the PR is not to exit if qemu-agent is not running in the VM, which might be the case for some images (e.g. non-openqa images)

but on next call to execute_cmd we will exit anyway , right ? So do you want to exit if you not able to execute_cmd or not ?
Also important point to call things what they really are. And I have some doubts that only reason why you can not execute command is the fact that agent is not running. The only thing which you know for sure when your command timeout is that command timeout :)

**Comment by asmorodskyi** [comment]
If command fail to execute we should fail . You trying to solve the problem which suppose to be solved inside test not inside qatrfm-lib.
Only thing which we could provide from framework perspective is provide some dedicated function something "check_agent" which will not raise exception but just return if agent is running or not. And then it is test writer headache to check in test level "Do we have agent running ? " and what to do with that 

But framework perspective it is clear for me - if command fails we failing

File: qatrfm/domain.py (hunk 1 above)
Comment (jlausuch) [reply qatrfm/domain.py]: >  So do you want to exit if you not able to execute_cmd or not ?

I don't want to exit at deployment time (= qemu-agent is not mandatory for an image). But in the test, that should be known and the user who writes the test will know if execute_cmd can be used or not.

**Comment by jlausuch** [comment]
I will do a small change, just found out about `{"execute":"guest-ping"} `command.

**Comment by jlausuch** [comment]
with the new approach: https://pastebin.com/raw/v4zgqB5k

**Comment by jlausuch** [comment]
Proof runs:

- with IP without qemu-agent, try to execute_cmd: https://pastebin.com/raw/fSm2Dy6p
- with IP and enabling qemu-agent manually in the VM via execute_ssh_cmd: https://pastebin.com/raw/w99PFVDn
- without IP and without qemu-agent: https://pastebin.com/raw/jPHNy1Qe
- without IP and with qemu-agent enabled: https://pastebin.com/raw/AjqaD3ni

**Pull request merged**